### ISSUE TITLE
Issue: Can sudo force a password to ensure gsudo security

### ISSUE THREAD

**Issue by loobug** [gerardog/gsudo#147]
Issue: Can sudo force a password to ensure gsudo security

**Comment by gerardog** [comment]
May I please ask you to edit your question, remove all the content from issue-template and just explain your question a little bit? Thanks

If your question is regarding skipping the UAC popup, and instead asking for your sudo password in plain text, I refrained from that as it would be less secure. 

Two issues with implementing this:
- Your password could be captured by a keylogger. 
- To skip the UAC popup, the credentials cache would have to be started all the time, which is an additional security concern. See [Security Considerations doc](https://gerardog.github.io/gsudo/docs/security#what-are-the-risks-of-running-gsudo)
Keep in mind that due to windows vs linux/unix differences, there is no known way to make gsudo as secure as the original sudo.

 `gsudo` is a convenience tool and there is always a tradeoff, but it's hard to set a limit that makes everyone happy. 

/dupe #118

**Comment by gerardog** [comment]
@loobug, can you explain your context, motivations for this ask, why the current model doesn´t work for you, please?

**Comment by loobug** [comment]
I am worried that my computer can be easily escalated by remote control, so I want to increase the difficulty, add a similar privileged password, do not need an operating system, only need to add a security line to gsudo, I hope you can accept it，thank you!

**Comment by gerardog** [comment]
Best way to reduce the attack vector is to not enable the credentials cache. The default is CacheMode Explicit, which means that the cache won't ever start, unless you explicitly run `gsudo cache on`. You can reset back to that default with `gsudo config cachemode --reset`

Or do you mean to have an active cache, but still let gsudo ask for a password? It can't be on the console window, since that allows a non-elevated process to keylog your password. That would be only a false sense of security.

gsudo might create an isolated secure desktop (i.e. the UAC popup runs on a secure desktop)... but how is that better than accepting the UAC popup every time?

**Comment by loobug** [comment]
ok, i configured windows group policy to make UAC enter the credentials every time the popup window, do you think this will be safer?

**Comment by gerardog** [comment]
It depends on whom are you protecting from. 

When you are logged in as an admin, UAC turns into a convenience Windows feature. ([here I explain what that means](https://gerardog.github.io/gsudo/docs/security)) It will just prevent you or other desktop users from running something as admin inadvertently. By default, if your user is admin, it would just ask Yes/No. If you leave the computer unattended, anyone can click Yes. By asking for a password on the popup, then they would need to know and type the password to elevate. AFAICT, this only protects you from other humans.

If you leave your computer logged in and unattended, setting it to always ask for credentials will protect you from someone with access to your keyboard, provided that you haven't left elevated windows running. But won't make a difference for viruses/malware. Those will still bypass UAC by hacking Windows or gsudo, and only Microsoft Defender or other AV will protect you. There are more than a hundred UAC bypass techniques disclosed, and many of them still work today.

I personally use UAC in always notify (which reduces the Windows attack vector just a little bit), have auto-lock in place when idle, but don't ask for my password every single time:
![image](https://user-images.githubusercontent.com/3901474/183329536-1134691b-3d62-41d4-8df5-255cdcdcd88f.png)

Bottom line is, 'ask for credentials' won't help you from malware. Setting "Always notify" may (or not) help a bit. Better alternative: create an admin user, then remove admin privileges from your user (that will also make you type your credentials each time). Best one: never elevate in the same desktop (log-out user, then log in admin, or use "switch accounts"). 

There is a tradeoff between security and usability. Using gsudo is less secure than not using it. Today I can't figure out how adding a password could change that.

**Comment by loobug** [comment]
think you, You are very patient and professional。

**Comment by gerardog** [comment]
You are welcome @loobug.
